presto-hive/src/main/java/com/facebook/presto/hive/HivePageSourceProvider.java

```diff
@@ -154,17 +161,24 @@ public ConnectorPageSource createPageSource(ConnectorTransactionHandle transacti
                     start,
                     length,
                     schema,
-                    regularColumns,
+                    doCoercion ? regularCoercedColumnHandles : regularOriginalColumnHandles,
```

Comment:
replace this with `extractRegularColumnHandles(regularColumnMappings, doCoercion)`